### PR TITLE
add new option "POWERLEVEL9K_BATTERY_HIDE_ABOVE_THRESHOLD"

### PR DESCRIPTION
**Pull request by sambadevi**
Hides the battery segment if percentage hits 100%

### CODE REVIEW

**Comment by sambadevi** [comment]
If you add `POWERLEVEL9K_BATTERY_HIDE_FULL=true` to your p9k config the battery segment will be hidden if the percentage is equal to 100%

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -488,14 +488,27 @@ prompt_battery() {
     fi
   fi
 
-  # override the default color if we are using a color level array
-  if [[ -n "$POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND" ]] && [[ "${(t)POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND}" =~ "array" ]]; then
-    local segment=$(( 100.0 / (${#POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND} - 1 ) ))
-    local offset=$(( ($bat_percent / $segment) + 1 ))
-    "$1_prompt_segment" "$0_${current_state}" "$2" "${POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND[$offset]}" "${battery_states[$current_state]}" "${message}" "BATTERY_ICON"
+  if [[ -v "POWERLEVEL9K_BATTERY_HIDE_FULL" && "$POWERLEVEL9K_BATTERY_HIDE_FULL" == true ]]; then
+      if [[ "$bat_percent" != "100" ]]; then
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: Wouldn't it be easier to inverse that condition and exit early?
Something like this:
```zsh
if [[ "${POWERLEVEL9K_BATTERY_HIDE_FULL}" == "true" && "${bat_percent}" == 100 ]]; then
  return
fi
```
That way we get rid of that indentations and the code duplication..

**Comment by dritter** [comment]
Btw. Something itches my head: I always try to ~~reduce the amount of variables~~ make variables more  meaningful in P9K. So it might be worth it to rename that variable to `POWERLEVEL9K_BATTERY_HIDE_ABOVE_THRESHOLD` and make the value an integer. That way we would gain more flexibility without making the code more complex..

What do you think?

**Comment by sambadevi** [comment]
Thanks! I didn’t think about the code duplication. I will take a look at this later on and update the PR accordingly

File: powerlevel9k.zsh-theme [hunk 2]
```diff
@@ -488,28 +488,19 @@ prompt_battery() {
...
-      # override the default color if we are using a color level array
-      if [[ -n "$POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND" ]] && [[ "${(t)POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND}" =~ "array" ]]; then
-        local segment=$(( 100.0 / (${#POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND} - 1 ) ))
-        local offset=$(( ($bat_percent / $segment) + 1 ))
-        "$1_prompt_segment" "$0_${current_state}" "$2" "${POWERLEVEL9K_BATTERY_LEVEL_BACKGROUND[$offset]}" "${battery_states[$current_state]}" "${message}" "BATTERY_ICON"
-      else
-        # Draw the prompt_segment
-        "$1_prompt_segment" "$0_${current_state}" "$2" "${DEFAULT_COLOR}" "${battery_states[$current_state]}" "${message}" "BATTERY_ICON"
-      fi
+  if [[ -v "POWERLEVEL9K_BATTERY_HIDE_ABOVE_THRESHOLD" && "${bat_percent}" -ge $POWERLEVEL9K_BATTERY_HIDE_ABOVE_THRESHOLD ]]; then
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: Could you add a comment before that line? Something like:
"Check if POWERLEVEL9K_BATTERY_HIDE_ABOVE_THRESHOLD is set and is exceeded by bat_percent, early exit."

Comment (sambadevi) [reply powerlevel9k.zsh-theme]: Sure, I’ll add the comment

**Pull request merged**